### ISSUE TITLE
FR: Add support for a hard-coded or default `--default-repo`

### ISSUE THREAD

**Issue by dhodun** [GoogleContainerTools/skaffold#7913]
tldr: we are looking for an easy way to set a GCR repo for a full profile that does not involved developers always having to add the `--default-repo` flag.

*************

We are hoping for either a value somewhere in the skaffold configs, or a skaffold-env file to automatically set basic repo or project-wide values, namely `--default-repo` to control which GCR repo is used for a particular profile. This allows us to remove hard-coded GCR repos.

Some potential options:
- skaffold-env file at the repo level
- setting this variable once based on kubeContext activation
- some sort of set-once global skaffold variable in the root yaml

The minikube to GKE developer experience right now is well served with the kubeContext activations - when a developer has minikube started and runs `skaffold dev`, everything is deployed locally. When they run `kubectx my_cluster` and then `skaffold dev` again, it will deploy to the new cluster. However, to achieve this simplicity, we end up hard-coding the GCR repo of our dev project in all of our `dev` profiles.

Minikube (base config):
https://github.com/GoogleCloudPlatform/solutions-template/blob/47bd57a97935277801eb28dfff630bfd63051b0c/microservices/sample_service/skaffold.yaml#L13-L17

GKE (dev profile):
https://github.com/GoogleCloudPlatform/solutions-template/blob/47bd57a97935277801eb28dfff630bfd63051b0c/microservices/sample_service/skaffold.yaml#L71-L78

We have avoided hard-coding our GCR repo in other profiles (prod etc), but ultimately it has caused issues because our most tested skaffold config is the `dev` profile and when we start moving new features into the `prod` profile, it often breaks. When we copy+paste into prod we often get non-obvious errors from remnants of the hard coding.

Thus far we have left in the hard-coding to help new developers, since they don't have to remember to add `--default-repo` every time they run `skaffold`.

### Docker build caveat

Another small caveat where (I think) we benefit from using hard-coding is feeding CloudBuild or Docker builds images to cache from:
https://github.com/GoogleCloudPlatform/solutions-template/blob/47bd57a97935277801eb28dfff630bfd63051b0c/microservices/sample_service/skaffold.yaml#L73-L75

I believe `skaffold` used to and may still treat these values as literals, so `--default-repo` wasn't respected here and the caching broke. It's possible, though, I may be thinking of the just the git tags.

### Information

- Skaffold version: v1.36
- Operating system: Mac / linux

**Comment by dhodun** [comment]
Upon further investigation it seems that if you run the following with the proper `kubeContext` already set, you can create a permanent global config that will connect a cluster with a GCR repo:

1. Set kubeContext to your dev cluster
2. run: `skaffold config set default-repo "$SKAFFOLD_DEFAULT_REPO"`

You now have a skaffold config that looks something like this:

```
global:
  survey:
    last-prompted: "2022-11-28T13:29:56-05:00"
  collect-metrics: true
  update:
    last-prompted: "2022-11-28T13:40:51-05:00"
kubeContexts:
- kube-context: gke_YOUR_PROJECT_ID_us-central1_YOUR_CLUSTER_NAME
  default-repo: gcr.io/SKAFFOLD_DEFAULT_REPO
  ```

This may be a short term workaround for us, but still involves intervention on developer environment setup, and is a bit non-obvious what is happening moving forward.

**Comment by aaron-prindle** [comment]
Closing as a dupe of #1545 1545
